core/src/main/java/dev/streamx/cli/license/LicenseFetcher.java

```diff
@@ -0,0 +1,110 @@
+package dev.streamx.cli.license;
+
+import static dev.streamx.cli.util.ExceptionUtils.sneakyThrow;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.streamx.cli.exception.LicenseException;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jetbrains.annotations.NotNull;
+
+@ApplicationScoped
+class LicenseFetcher {
+
+  @ConfigProperty(name = "streamx.cli.license.current-license-url")
```

Comment:
Is it build time property or it can be set using env variable in run time?

Reply:
That's good catch